src/lib/onlyText.ts

```diff
@@ -22,9 +21,14 @@ const onlyText = (children: ReactNode | ReactNode[]): string => {
   return Children.toArray(children).reduce((text: string, child: ReactNode): string => {
     let newText = '';
 
-    if (isValidElement(child) && hasChildren(child)) {
+    const isValidElementResult = isValidElement<{ children?: React.ReactNode | React.ReactNode[] }>(
```

Comment:
Why is it necessary to specify the type here? `<{ children?: React.ReactNode | React.ReactNode[] }>`

Reply:
For say to TS that we are expecting `children` property in `child.props`.

<img width="857" alt="image" src="https://github.com/fernandopasik/react-children-utilities/assets/5850354/4cb25d38-79da-4829-83a3-99bf89907bb0">

_[link to TS playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBDAnmApnA3nASighgYxgDkIATNAXzgDMoIQ4ByKPQxgbgChRJYM4AwgAtgAG1IsAdgBo4wAM4A1XKOCkAoqJQgUk+FVr0mLAjA6dOKAB694+CJPl2R4gCoQAyjCjBJAczgAXjgACnwXUgB+AC5sVmIyFABKWKcffyCAPgxOODlqUKRUCALwsVIgwODGAFdJcmpfFFJGOAAfNrgy8UrgyRrRUXbOopQSrojepgAjCAgtXElGJJy8vJYYGqhJJnM8igs84AKQgCkPAHkiADo03z9jxDCIlarq9Apl1bWNrZ3GPZwA65OC-bahboVXDyOD9EDTFBQdpwO7+JLXGCebz3EJJLgULiceyOeAOUSIVzWeDBZ7lKSxHCmEjkZGMwjMlAAbQAuikUdiMoFsugQcdQgBCWniKRyEmLfBjAoAQSgUFwiBWADJNXBxQplKoNFodHopRJdEkViK1qCUJtwZD3F50n4zVI8SDges7X9BBEpBiICq1U9Ie7rixSDUFSEQjAqakBX5ZJCGfEOXzUQEhd88lp4JIUAB3SlWam7Qk24lOORKFRqTTaXQwHDyAbl-X1o1N02Qj1Vhw1oTQ4R03RBWsGhvG5ut9twbVMMO6Vq+Cbla5gOhgeSVtZikKdw2Nk0tlBt0TwRfD+Sj6UW3NrQslqkTskUqlmzfb+TXZeSfs1ioFBRHkNADyPacezPC8rx1cUbzvc0AMfPJn1LcsAS4G1gNAtBrRtWFiwwidHSxF0zUA-ZDh+H1wXjMs-wcfBcBgEJ0KpQCKFkAEPQJCxrFsOAGlwed3ww9ggA)_